src/searching/linear_search.rs

```diff
@@ -0,0 +1,131 @@
+/// ## Linear Search algorithm.
+/// Iterating over the array linearly and returns the index value of the target element in the array.
+///
+/// Time complexcity = O(N/2) where `N` is the length of the array.
```

Comment:
This is not true. The lower bound of linear search is the `N` elements, so time complexity is `O(N)` in the worst case.

You can think of it this way: the worst case is that the element is not found since we would have to iterate over the whole collection.

I see that you implemented a search that iterates simultaneously from the left and the right until the middle. The problem with this solution is that it is still `O(N)`. The explanation goes a bit beyond the scope of a PR review, but the reasons are twofold:

* `O(N / 2) = O(N)` -> `1/2` is a constant, so it doesn't really affect the "nature" of the function. The amount of meaningful operations the algorithm does increases linearly with respect to its input `N`.
* You are not taking into account the amount of *atomic* operations your algorithm does. If we took comparisons as an example of an atomic operation, your implementation does twice the amount of comparisons that the simpler linear search implementation does, so it really is more like `O(2 * N / 2)` and not `O(N / 2)`.

The code looks fine, but all of the above is relevant since we don't want "clever" code, we want "educational" code.